devtools/utils.py

```diff
@@ -149,12 +149,19 @@ class DataClassType(metaclass=MetaDataClassType):
 
 class MetaSQLAlchemyClassType(type):
     def __instancecheck__(self, instance: 'Any') -> bool:
+        try:
```

Comment:
surely neater to do

```py
    # DeclarativeBase exists for SQLAlchemy>=2.0
    try:
        from sqlalchemy.orm import DeclarativeBase  # type: ignore
    except ImportError:
        pass
    else:
        return isinstance(instance, DeclarativeBase)

    # DeclarativeMeta exists for older SQLAlchemy
    try:
        from sqlalchemy.ext.declarative import DeclarativeMeta  # type: ignore
    except ImportError:
        return False
    else:
        return isinstance(instance.__class__, DeclarativeMeta)
```

???

Reply:
this should also minimise the change footprint which is nice.

Reply:
the thing is that DeclarativeMeta exists also in sqlalchemy2 and it possible to declare the model the old way as well (using declarative_base()). 

So the old way works in sqlalchemy2 and sqlalchemy1
```
from sqlalchemy.ext.declarative import declarative_base
Base = declarative_base()
class User(Base):
    ...
```

The new way is only in sqlalchemy2:
```
from sqlalchemy.orm import DeclarativeBase
class Base(DeclarativeBase):
    ...

class User(Base):
    ...
```

Your proposed change will successfully import DeclarativeBase from sqlalchemy2 and fail checking instance against DeclarativeBase if it was declared the old way (using declarative_base()). 

--------------------------------

BTW. I came up with an another addition. Currently debug() fails displaying sqlalchemy instance if some attributes were deferred. It happens in the async mode when you do getattr on the such fields with the error:
```
    !!! error pretty printing value: MissingGreenlet("greenlet_spawn has not been called; can't call await_only() here. Was IO attempted in an unexpected place?")
```
That makes debug() useless in the async mode. I have a patch that marks columns as "deferred" instead of trying to load it.
Do you need it? Should it be a part of another MR?

Reply:
deferred  fields: I think adding that to the same PR is fine, it's only a small change.

On the main issue - I'm stuggling to see how my code is fundamentally different to yours - they both do "if importing this was successfuly, check against it, otherwise try this other import".

Happy to switch around the order if you think it would be better to try `DeclarativeMeta` first.

Reply:
What you proposed is different to mine. This way it will work. The difference is that it must proceed to the second try-else block after the `isinstance(instance, DeclarativeBase)` returns false. In this case we save time by skipping importing `DeclarativeMeta` if a model is defined the new way.

```        try:
            from sqlalchemy.orm import DeclarativeBase  # type: ignore
        except ImportError:
            pass
        else:
            if isinstance(instance, DeclarativeBase):
                return True

        try:
            from sqlalchemy.ext.declarative import DeclarativeMeta  # type: ignore
        except ImportError:
            pass
        else:
            return isinstance(instance.__class__, DeclarativeMeta)```


With the "deferred" thingy I have a dilemma.
currently it displays something like:
```
Topology(
    contained_by='<deferred>',
    contains='<deferred>',
    contains_singular='<deferred>',
    id=7,
    label='<deferred>',
    name='<deferred>'
)
```
But we can also skip the deferred fields and show only the loaded ones, like:
```
Topology(
    id=7
)
```
Personally I don't care. What is your opinion?